zipline/algorithm.py

```diff
@@ -1063,7 +1063,7 @@ def fetch_csv(self,
 
         return csv_data_source
 
-    def add_event(self, rule=None, callback=None):
+    def add_event(self, rule, callback=None):
```

Comment:
Is there a reason `callback` is allowed to be None here?  I would think failing to pass a callback is almost always a bug.

Reply:
I'm guessing it was convenient for tests. Maybe @llllllllll knows more.

Reply:
Yeah, I don't think there are good defaults for either. We should make them required.